scale/batch/models.py

```diff
@@ -0,0 +1,189 @@
+        :param title: The human-readable name of the batch
+        :type title: string
+        :param description: An optional description of the batch
+        :type description: string
+        :returns: The newly created batch
+        :rtype: :class:`batch.models.Batch`
+
+        :raises :class:`batch.exceptions.BatchError`: If general batch parameters are invalid
+        """
+
+        # Attempt to get the batch job type
+        try:
+            job_type = JobType.objects.filter(name='scale-batch').last()
+        except JobType.DoesNotExist:
+            raise BatchError('Missing required job type: scale-batch')
+
+        # Create an event to represent this request
+        trigger_desc = {'user': 'Anonymous'}
+        event = TriggerEvent.objects.create_trigger_event('USER', None, trigger_desc, timezone.now())
+
+        batch = Batch()
+        batch.title = title
+        batch.description = description
+        batch.recipe_type = recipe_type
+        batch.definition = definition.get_dict()
+        batch.event = event
+        batch.save()
+
+        # Setup the job data to process the batch
+        data = JobData()
+        data.add_property_input('Batch ID', str(batch.id))
+
+        # Schedule the batch job
+        job = Queue.objects.queue_new_job(job_type, data, event)
+        batch.job = job
+        batch.save()
+
+        return batch
+
+
+class Batch(models.Model):
+    """Represents a batch of historical recipes to be re-processed on the cluster
+
+    :keyword title: The human-readable name of the batch
+    :type title: :class:`django.db.models.CharField`
+    :keyword description: An optional description of the batch
+    :type description: :class:`django.db.models.TextField`
+    :keyword status: The status of the batch
+    :type status: :class:`django.db.models.CharField`
+
+    :keyword recipe_type: The type of recipe to re-process
+    :type recipe_type: :class:`django.db.models.ForeignKey`
+    :keyword event: The event that triggered the creation of this batch
+    :type event: :class:`django.db.models.ForeignKey`
+    :keyword job: The job that will setup the batch and schedule jobs for re-processing
+    :type job: :class:`django.db.models.ForeignKey`
+
+    :keyword definition: JSON definition for setting up the batch
+    :type definition: :class:`djorm_pgjson.fields.JSONField`
+
+    :keyword created: When the batch was created
+    :type created: :class:`django.db.models.DateTimeField`
+    :keyword completed: When every job in the batch was completed successfully
+    :type completed: :class:`django.db.models.DateTimeField`
+    :keyword last_modified: When the batch was last modified
+    :type last_modified: :class:`django.db.models.DateTimeField`
+    """
+
+    BATCH_STATUSES = (
+        ('PENDING', 'PENDING'),
+        ('QUEUED', 'QUEUED'),
+        ('RUNNING', 'RUNNING'),
+        ('FAILED', 'FAILED'),
+        ('COMPLETED', 'COMPLETED'),
+        ('CANCELED', 'CANCELED'),
+    )
```

Comment:
I would have two statuses: SUBMITTED and CREATED

SUBMITTED is when you initially put in the batch
CREATED is once the creator job is completed and the batch has been created